cdap-watchdog/src/main/java/co/cask/cdap/logging/write/AvroFileWriter.java

```diff
@@ -174,9 +182,25 @@ private AvroFile createAvroFile(LoggingContext loggingContext, long timestamp) t
     return avroFile;
   }
 
-  private Location createLocation(String pathFragment, long timestamp) throws IOException {
+  private Location createLocation(LoggingContext loggingContext, long timestamp) throws IOException {
+    Map<String, LoggingContext.SystemTag> systemTags = loggingContext.getSystemTagsMap();
+    LoggingContext.SystemTag topLevelTag;
+    if (systemTags.containsKey(NamespaceLoggingContext.TAG_NAMESPACE_ID)) {
+      topLevelTag = systemTags.get(NamespaceLoggingContext.TAG_NAMESPACE_ID);
+    } else if (systemTags.containsKey(SystemLoggingContext.TAG_SYSTEM_ID)) {
+      topLevelTag = systemTags.get(SystemLoggingContext.TAG_SYSTEM_ID);
+    } else {
+      throw new IllegalArgumentException(
```

Comment:
Not a good idea to throw exception from here. A stray log event will block the whole pipeline. Instead we should validate the log event when it is received, and drop it if it does not have right tags.